scopesim/server/database.py

```diff
@@ -170,38 +222,50 @@ def download_package(pkg_path, save_dir=None, url=None, from_cache=None):
         The absolute path to the saved ``.zip`` package
 
     """
-    if isinstance(pkg_path, (list, tuple)):
-        save_path = [download_package(pkg, save_dir, url) for pkg in pkg_path]
+    # todo: add proper depreciation warning
+    text = "Function Depreciated --> please use scopesim.download_package-s-()"
+    logging.warning(text)
+    print(text)
 
-    elif isinstance(pkg_path, str):
-        if pkg_path[-4:] != ".zip":
-            logging.warning("Appended '.zip' to %s", pkg_path)
-            pkg_path += ".zip"
+    pkg_names = [pkg.split("/")[-1].replace(".zip", "") for pkg in pkg_path]
+    return download_packages(pkg_names, release="stable", save_dir=save_dir,
+                             from_cache=from_cache)
 
-        if url is None:
-            url = rc.__config__["!SIM.file.server_base_url"]
-        if save_dir is None:
-            save_dir = rc.__config__["!SIM.file.local_packages_path"]
+def get_server_elements(url, unique_str="/"):
+    """
+    Returns a list of file and/or directory paths on the HTTP server ``url``
 
-        if not os.path.exists(save_dir):
-            os.mkdir(save_dir)
+    Parameters
+    ----------
+    url : str
+        The URL of the IRDB HTTP server.
 
-        try:
-            if from_cache is None:
-                from_cache = rc.__config__["!SIM.file.use_cached_downloads"]
-            cache_path = download_file(url + pkg_path, cache=from_cache)
-            save_path = os.path.join(save_dir, os.path.basename(pkg_path))
-            file_path = shutil.copy2(cache_path, save_path)
+    unique_str : str, list
+        A unique string to look for in the beautiful HTML soup:
+        "/" for directories this, ".zip" for packages
 
-            with zipfile.ZipFile(file_path, 'r') as zip_ref:
-                zip_ref.extractall(save_dir)
+    Returns
+    -------
+    paths : list
+        List of paths containing in ``url`` which contain ``unique_str``
 
-        except HTTPError:
-            ValueError(f"Unable to find file: {url + pkg_path}")
+    """
+    if isinstance(unique_str, str):
+        unique_str = [unique_str]
 
-        save_path = os.path.abspath(save_path)
+    try:
+        result = requests.get(url).content
+    except Exception as error:
+        raise ValueError(f"URL returned error: {url}") from error
```

Comment:
What! `raise ... from error`? Never seen that, this is useful. I'll update my other suggestions to reflect this.